js/viz/core/renderers/renderer.js

```diff
@@ -1946,19 +1949,36 @@ Renderer.prototype = {
         return elem.attr({ text: text, x: x || 0, y: y || 0 });
     },
 
-    linearGradient: function(stops) {
-        const id = getNextDefsSvgId();
-        const that = this;
-        const gradient = that._createElement('linearGradient', { id: id }).append(that._defs);
+    linearGradient: function(stops, id = getNextDefsSvgId(), rotationAngle) {
+        const gradient = this._createElement('linearGradient', {
+            id: id,
+            gradientTransform: `rotate(${rotationAngle || 0})`
+        }).append(this._defs);
         gradient.id = id;
 
-        stops.forEach((stop) => {
-            that._createElement('stop', { offset: stop.offset, 'stop-color': stop['stop-color'] }).append(gradient);
-        });
+        this._createGradientStops(stops, gradient);
+
+        return gradient;
+    },
+
+    radialGradient: function(stops, id) {
+        const gradient = this._createElement('radialGradient', { id: id }).append(this._defs);
+
+        this._createGradientStops(stops, gradient);
 
         return gradient;
     },
 
+    _createGradientStops: function(stops, group) {
+        stops.forEach((stop) => {
+            this._createElement('stop', {
+                offset: stop.offset,
+                'stop-color': stop['stop-color'] || stop.color,
```

Comment:
```suggestion
                'stop-color': stop['stop-color'] ?? stop.color,
```
Cannot `stop['stop-color']` be a `0`?